plugins/modules/system/sap_task_list_execute.py

```diff
@@ -0,0 +1,437 @@
+            # treat like dict - we assume that if the first two tags
+            # in a series are different, then they are all different.
+            if len(element) == 1 or element[0].tag != element[1].tag:
+                aDict = xml_dict(element)
+            # treat like list - if the first two tags
+            # in a series are the same, then the rest are the same.
+            else:
+                for list_return in element:
+                    if list_return:
+                        # treat like dict
+                        if len(list_return) == 1 or list_return[0].tag != list_return[1].tag:
+                            list_out.append(xml_dict(list_return))
+                        # treat like list
+                        elif list_return[0].tag == list_return[1].tag:
+                            list_out.append(list_out.xml_list(list_return))
+                    elif list_return.text:
+                        text = list_return.text.strip()
+                        if text:
+                            list_out.append(text)
+                # put the list in dictionary;
+                # the key is the tag name the list elements all share in common
+                # the value is the list itself
+                aDict = {element[0].tag: list_out}
+            # if the tag has attributes, add those to the dict
+            if element.items():
+                aDict.update(dict(element.items()))
+            dict_out.update({element.tag: aDict})
+        elif element.items():
+            dict_out.update({element.tag: dict(element.items())})
+        # if there are no child tags and no attributes, extract the text
+        else:
+            dict_out.update({element.tag: element.text})
+    return dict_out
+
+
+def call_rfc_method(connection, method_name, kwargs):
+    # PyRFC call function
+    resp = connection.call(method_name, **kwargs)
+
+    return resp
+
+
+def process_exec_settings(task_settings):
+    # processes task settings to objects
+    exec_settings = {}
+    for settings in task_settings:
+        temp_dict = {settings.upper(): 'X'}
+        for key, value in temp_dict.items():
+            exec_settings[key] = value
+    return exec_settings
+
+
+def process_task_parameters(task_parameters):
+    parameter_list = []
+    raw_task_list = []
+    task_list = []
+    if task_parameters is not None:
+        for parameters in task_parameters:
+            if parameters[0].startswith('CL_'):
+                length = len(parameters)
+                raw_task_list = raw_task_list + [{'TASKNAME': parameters[0]}]
+                task_list = [i for n, i in enumerate(raw_task_list) if i not in raw_task_list[n + 1:]]
+                if length == 3:
+                    parameter_list = parameter_list + [{'TASKNAME': parameters[0], 'FIELDNAME': parameters[1], 'VALUE': parameters[2]}]
+            else:
+                return False
+    #
+    return [{'parameter_list': parameter_list}, {'task_list': task_list}]
```

Comment:
Is there a requirement for the order of the tasks in `task_list`? If not, instead of the `raw_task_list` maneuver up there, you might add all task names to a `set`, then `task_list = [{'TASKNAME': x} for x in task_set]`.

Also, I cannot help thinking that:
```python
parameter_list = [
    zip(['TASKNAME', 'FILEDNAME', 'VALUE'], params)
    for params in task_parameters
    if len(params) == 3 and params[0].startswith('CL_')
]
```

Reply:
if you perform a validation on all the tasknames previously (for the `CL_` prefix), then it can become closer to the output this code produce today.

Reply:
obsolete because this function is removed